# DEVOIR 3 NICOLAS ST-GERMAIN

In [1]:
%matplotlib inline
import pandas as pd
pd.set_option('display.max_rows', 30)
pd.set_option('display.max_columns', 500)
pd.set_option('display.width', 1000)
pd.options.display.float_format = "{:.2f}".format

J'ai importé les modules nécessaires pour le devoir et ajouté des options pour y afficher toutes les colonnes et améliorer l'affiche des décimales pour les chiffres.

In [2]:
subs_contrat = pd.read_csv("grants.csv", low_memory=False)

Étant donné que le fichier est trop volumineux, il a fallu ajouté "low_memory=False" pour réussir à l'ouvrir.

In [3]:
subs_contrat

ref_number  amendment_number amendment_date agreement_type recipient_type recipient_business_number                               recipient_legal_name                           recipient_operating_name research_organization_name recipient_country recipient_province                                     recipient_city recipient_postal_code                            federal_riding_name_en                            federal_riding_name_fr  federal_riding_number                                       prog_name_en                                       prog_name_fr                                    prog_purpose_en                                    prog_purpose_fr                                 agreement_title_en                                 agreement_title_fr    agreement_number agreement_value foreign_currency_type foreign_currency_value agreement_start_date agreement_end_date coverage                                     description_en  \
0                GC-2017-Q1-00001                 0            NaN              C            NaN                       NaN                Agriculture in the Classroom Canada                                                NaN                        NaN                CA                NaN                               Headingley, Manitoba                   NaN                                               NaN                                               NaN                    nan  AgriCompetitiveness: Fostering Business Develo...  Agri-compétitivité: Développement des entreprises  AgriCompetitiveness Program -Fostering Busines...  Programme Agri-compétitivité - Volet Favoriser...  Fostering Youth Development and Leadership to ...  Favoriser l'épanouissement et le leadership de...         GF2-FBD-009     $567,786.00                   NaN                    NaN           2017-05-23                NaN      NaN                                                NaN   
1                GC-2017-Q1-00002                 0            NaN              C            NaN                       NaN                                 Amra Services Inc.                                                NaN                        NaN                CA                NaN  Coquitlam, British Columbia|Coquitlam, Colombi...                   NaN                                               NaN                                               NaN                    nan                  AgriMarketing: Market Development          Agri-marketing: Développement des marchés  AgriMarketing Program - Market Development (AM...  Programme Agri-marketing - Volet Développement...            Glutenull Export and Market Development  Développement des exportations et des marchés ...          GF2-MD-757      $39,240.00                   NaN                    NaN           2017-05-18                NaN      NaN                                                NaN   
2                GC-2017-Q1-00003                 0            NaN              C            NaN                       NaN                            Atlantic Grains Council                                                NaN                        NaN                CA                NaN  Moncton, New Brunswick|Moncton, Nouveau  Bruns...                   NaN                                               NaN                                               NaN                    nan                 Agri-innovation: IND-LED - R and D  Volet Recherche et développement dirigé par l'...  Agri-Innovations Program -Industry Led Researc...  Programme Agri-innovation - Volet Activités de...  Focused Research for Atlantic Grains and Oilse...  Recherche axée sur les besoins des producteurs...    AIP-P264 Amnd #1     $761,657.00                   NaN                    NaN           2017-04-20                NaN      NaN                                                NaN   
3                GC-2017-Q1-00004                 0            NaN              C            NaN                       NaN 

L'affichage de subs_contrat permet de visualiser ce qu'il en tient de ce fichier.

In [4]:
subs_contrat.dtypes

ref_number                     object
amendment_number                int64
amendment_date                 object
agreement_type                 object
recipient_type                 object
recipient_business_number      object
recipient_legal_name           object
recipient_operating_name       object
research_organization_name     object
recipient_country              object
recipient_province             object
recipient_city                 object
recipient_postal_code          object
federal_riding_name_en         object
federal_riding_name_fr         object
                               ...   
agreement_value                object
foreign_currency_type          object
foreign_currency_value         object
agreement_start_date           object
agreement_end_date             object
coverage                       object
description_en                 object
description_fr                 object
naics_identifier              float64
expected_results_en            object
expected_res

le .dtypes permet de savoir le type des différentes colonnes.

In [5]:
subs_contrat.columns

Index(['ref_number', 'amendment_number', 'amendment_date', 'agreement_type', 'recipient_type', 'recipient_business_number', 'recipient_legal_name', 'recipient_operating_name', 'research_organization_name', 'recipient_country', 'recipient_province', 'recipient_city', 'recipient_postal_code', 'federal_riding_name_en', 'federal_riding_name_fr', 'federal_riding_number', 'prog_name_en', 'prog_name_fr', 'prog_purpose_en', 'prog_purpose_fr', 'agreement_title_en', 'agreement_title_fr', 'agreement_number', 'agreement_value', 'foreign_currency_type', 'foreign_currency_value', 'agreement_start_date', 'agreement_end_date', 'coverage', 'description_en', 'description_fr', 'naics_identifier', 'expected_results_en', 'expected_results_fr', 'additional_information_en', 'additional_information_fr', 'owner_org', 'owner_org_title'], dtype='object')

In [6]:
subs_contrat = subs_contrat.drop(['federal_riding_name_en', 'prog_name_en','prog_purpose_en','agreement_title_en','description_en','expected_results_en','additional_information_en'], 1)

En affichant les colonnes dans le IN, j'ai pu retirer ceux qui ne m'intéresse pas. Dans ce cas-ci, il s'agissait de ceux se terminant par (_en) ._. Donc je les ai retirés.

In [7]:
subs_contrat.columns

Index(['ref_number', 'amendment_number', 'amendment_date', 'agreement_type', 'recipient_type', 'recipient_business_number', 'recipient_legal_name', 'recipient_operating_name', 'research_organization_name', 'recipient_country', 'recipient_province', 'recipient_city', 'recipient_postal_code', 'federal_riding_name_fr', 'federal_riding_number', 'prog_name_fr', 'prog_purpose_fr', 'agreement_title_fr', 'agreement_number', 'agreement_value', 'foreign_currency_type', 'foreign_currency_value', 'agreement_start_date', 'agreement_end_date', 'coverage', 'description_fr', 'naics_identifier', 'expected_results_fr', 'additional_information_fr', 'owner_org', 'owner_org_title'], dtype='object')

In [8]:
subs_contrat.columns = ['numero_reference', 'numero_amendement', 'date_amendement', 'type_entente','type_beneficiaire', 'numero_compagnie_beneficiaire', 'nom_legal_beneficiaire','nom_operationnel_beneficiaire', 'nom_organisation_de_recherche','pays_beneficiaire', 'province_beneficiaire', 'ville_beneficiaire','code_postal_beneficiaire', 'nom_du_comte_federal','numero_du_comte_federal', 'nom_programme', 'but_du_programme','titre_entente', 'numero_entente', 'valeur_entente','type_monnaie_etrangeres', 'valeur_monaie_etrangere','date_debut_entente', 'date_fin_entente', 'couverture','description', 'identifiant_SCIAN', 'resultat_attendu','information_additionnelle', 'abbreviation_ministère', 'ministere_responsable']

In [9]:
subs_contrat

numero_reference  numero_amendement date_amendement type_entente type_beneficiaire numero_compagnie_beneficiaire                             nom_legal_beneficiaire                      nom_operationnel_beneficiaire nom_organisation_de_recherche pays_beneficiaire province_beneficiaire                                 ville_beneficiaire code_postal_beneficiaire                              nom_du_comte_federal  numero_du_comte_federal                                      nom_programme                                   but_du_programme                                      titre_entente      numero_entente valeur_entente type_monnaie_etrangeres valeur_monaie_etrangere date_debut_entente date_fin_entente couverture                                        description  identifiant_SCIAN                                   resultat_attendu                          information_additionnelle abbreviation_ministère                              ministere_responsable
0                GC-2017-Q1-00001                  0             NaN            C               NaN                           NaN                Agriculture in the Classroom Canada                                                NaN                           NaN                CA                   NaN                               Headingley, Manitoba                      NaN                                               NaN                      nan  Agri-compétitivité: Développement des entreprises  Programme Agri-compétitivité - Volet Favoriser...  Favoriser l'épanouissement et le leadership de...         GF2-FBD-009    $567,786.00                     NaN                     NaN         2017-05-23              NaN        NaN                                                NaN                nan                                                NaN                                                NaN               aafc-aac  Agriculture and Agri-Food Canada | Agriculture...
1                GC-2017-Q1-00002                  0             NaN            C               NaN                           NaN                                 Amra Services Inc.                                                NaN                           NaN                CA                   NaN  Coquitlam, British Columbia|Coquitlam, Colombi...                      NaN                                               NaN                      nan          Agri-marketing: Développement des marchés  Programme Agri-marketing - Volet Développement...  Développement des exportations et des marchés ...          GF2-MD-757     $39,240.00                     NaN                     NaN         2017-05-18              NaN        NaN                                                NaN                nan                                                NaN                                                NaN               aafc-aac  Agriculture and Agri-Food Canada | Agriculture...
2                GC-2017-Q1-00003                  0             NaN            C               NaN                           NaN                            Atlantic Grains Council                                                NaN                           NaN                CA                   NaN  Moncton, New Brunswick|Moncton, Nouveau  Bruns...                      NaN                                               NaN                      nan  Volet Recherche et développement dirigé par l'...  Programme Agri-innovation - Volet Activités de...  Recherche axée sur les besoins des producteurs...    AIP-P264 Amnd #1    $761,657.00                     NaN                     NaN         2017-04-20              NaN        NaN                                                NaN                nan                                                NaN  Cette entrée tient compte d'une augmentation d...               aafc-aac  Agriculture and Agri-Food Canada | Agriculture...
3                GC-2017-Q1-00004                  0             NaN            C        

Les trois IN ci-haut comportent les étapes que j'ai suivi pour traduire les colonnes restantes en français. Après les avoir affichées, j'ai changé les termes en français.

In [10]:
subs_contrat["valeur_entente"] = subs_contrat.valeur_entente.str.replace("$", "").str.replace(",","")

In [11]:
subs_contrat["valeur_entente"] = subs_contrat.valeur_entente.astype(float)

In [12]:
subs_contrat

numero_reference  numero_amendement date_amendement type_entente type_beneficiaire numero_compagnie_beneficiaire                             nom_legal_beneficiaire                      nom_operationnel_beneficiaire nom_organisation_de_recherche pays_beneficiaire province_beneficiaire                                 ville_beneficiaire code_postal_beneficiaire                              nom_du_comte_federal  numero_du_comte_federal                                      nom_programme                                   but_du_programme                                      titre_entente      numero_entente  valeur_entente type_monnaie_etrangeres valeur_monaie_etrangere date_debut_entente date_fin_entente couverture                                        description  identifiant_SCIAN                                   resultat_attendu                          information_additionnelle abbreviation_ministère                              ministere_responsable
0                GC-2017-Q1-00001                  0             NaN            C               NaN                           NaN                Agriculture in the Classroom Canada                                                NaN                           NaN                CA                   NaN                               Headingley, Manitoba                      NaN                                               NaN                      nan  Agri-compétitivité: Développement des entreprises  Programme Agri-compétitivité - Volet Favoriser...  Favoriser l'épanouissement et le leadership de...         GF2-FBD-009       567786.00                     NaN                     NaN         2017-05-23              NaN        NaN                                                NaN                nan                                                NaN                                                NaN               aafc-aac  Agriculture and Agri-Food Canada | Agriculture...
1                GC-2017-Q1-00002                  0             NaN            C               NaN                           NaN                                 Amra Services Inc.                                                NaN                           NaN                CA                   NaN  Coquitlam, British Columbia|Coquitlam, Colombi...                      NaN                                               NaN                      nan          Agri-marketing: Développement des marchés  Programme Agri-marketing - Volet Développement...  Développement des exportations et des marchés ...          GF2-MD-757        39240.00                     NaN                     NaN         2017-05-18              NaN        NaN                                                NaN                nan                                                NaN                                                NaN               aafc-aac  Agriculture and Agri-Food Canada | Agriculture...
2                GC-2017-Q1-00003                  0             NaN            C               NaN                           NaN                            Atlantic Grains Council                                                NaN                           NaN                CA                   NaN  Moncton, New Brunswick|Moncton, Nouveau  Bruns...                      NaN                                               NaN                      nan  Volet Recherche et développement dirigé par l'...  Programme Agri-innovation - Volet Activités de...  Recherche axée sur les besoins des producteurs...    AIP-P264 Amnd #1       761657.00                     NaN                     NaN         2017-04-20              NaN        NaN                                                NaN                nan                                                NaN  Cette entrée tient compte d'une augmentation d...               aafc-aac  Agriculture and Agri-Food Canada | Agriculture...
3                GC-2017-Q1-00004                  0             NaN            C    

Avec l'aide du web : https://www.ritchieng.com/pandas-changing-datatype/ , j'ai réussi à changer ma données "valeur_entente" en données lisible par Pandas. Pour ce faire, j'ai remplacé les symboles $ et , par des "" (donc rien) et ensuite j'ai changé le type de la variable, un peu comme nous faisions dans python avec le int() ou str() par exemple.

In [13]:
subs_contrat.dtypes

numero_reference                  object
numero_amendement                  int64
date_amendement                   object
type_entente                      object
type_beneficiaire                 object
numero_compagnie_beneficiaire     object
nom_legal_beneficiaire            object
nom_operationnel_beneficiaire     object
nom_organisation_de_recherche     object
pays_beneficiaire                 object
province_beneficiaire             object
ville_beneficiaire                object
code_postal_beneficiaire          object
nom_du_comte_federal              object
numero_du_comte_federal          float64
                                  ...   
but_du_programme                  object
titre_entente                     object
numero_entente                    object
valeur_entente                   float64
type_monnaie_etrangeres           object
valeur_monaie_etrangere           object
date_debut_entente                object
date_fin_entente                  object
couverture      

La preuve ci-haut : valeur_entente == float64

In [14]:
subs_contrat.groupby("ministere_responsable").valeur_entente.sum()

ministere_responsable
Agriculture and Agri-Food Canada | Agriculture et Agroalimentaire Canada                                                            1567055265.06
Canada Economic Development for Quebec Regions | Développement économique Canada pour les régions du Québec                         3815411324.00
Canada Revenue Agency | Agence du revenu du Canada                                                                                    15124758.00
Canadian Environmental Assessment Agency | Agence canadienne d'évaluation environnementale                                            42342369.06
Canadian Food Inspection Agency | Agence canadienne d'inspection des aliments                                                           810488.00
Canadian Heritage | Patrimoine canadien                                                                                            14541895641.62
Canadian Institutes of Health Research | Instituts de recherche en santé du Canada                    

En procédant comme ci-haut, j'ai réussi à voir où était le ministère de Pablo Rodriguez et quel était son appellation.

In [15]:
pCan = subs_contrat.ministere_responsable == "Canadian Heritage | Patrimoine canadien"

In [16]:
subs_contrat[pCan]

numero_reference  numero_amendement date_amendement type_entente type_beneficiaire numero_compagnie_beneficiaire                             nom_legal_beneficiaire nom_operationnel_beneficiaire nom_organisation_de_recherche pays_beneficiaire province_beneficiaire                                 ville_beneficiaire code_postal_beneficiaire nom_du_comte_federal  numero_du_comte_federal                                      nom_programme                                   but_du_programme                                      titre_entente numero_entente  valeur_entente type_monnaie_etrangeres valeur_monaie_etrangere date_debut_entente date_fin_entente couverture                                        description  identifiant_SCIAN                                   resultat_attendu                        information_additionnelle abbreviation_ministère                    ministere_responsable
165880          GC-2005-Q4-00001                  0             NaN            C               NaN                           NaN                       JAZZ EAST RISING ASSOCIATION                           NaN                           NaN                CA                   NaN      Halifax, Nova Scotia|Halifax, Nouvelle-Écosse                      NaN                  NaN                      nan                                                NaN  Fonds du Canada pour la présentation des arts ...                                                NaN            NaN       210000.00                     NaN                     NaN         2006-01-03              NaN        NaN                                                NaN                nan                                                NaN  Projet pluriannuel: du 2006-04-01 au 2009-03-31                    pch  Canadian Heritage | Patrimoine canadien
165881          GC-2005-Q4-00002                  0             NaN            C               NaN                           NaN                 CHILDREN'S FESTIVAL DE LA JEUNESSE                           NaN                           NaN                CA                   NaN                                    Ottawa, Ontario                      NaN                  NaN                      nan                                                NaN  Fonds du Canada pour la présentation des arts ...                                                NaN            NaN       246000.00                     NaN                     NaN         2006-01-03              NaN        NaN                                                NaN                nan                                                NaN  Projet pluriannuel: du 2006-04-01 au 2008-03-31                    pch  Canadian Heritage | Patrimoine canadien
165882          GC-2005-Q4-00003                  0             NaN            C               NaN                           NaN  OTTAWA CHAMBER MUSIC SOCIETY (THE) / LA SOCIÉT...                           NaN                           NaN                CA                   NaN                                    Ottawa, Ontario                      NaN                  NaN                      nan                                                NaN  Développement des compétences pour les organis...                                                NaN            NaN       112400.00                     NaN                     NaN         2006-01-03              NaN        NaN                                                NaN                nan                                                NaN  Projet pluriannuel: du 2005-11-01 au 2006-12-31                    pch  Canadian Heritage | Patrimoine canadien
165883          GC-2005-Q4-00004                  0             NaN            C               NaN                           NaN                                MACLAREN ART CENTRE                           NaN                           NaN                CA                   NaN                                    Barrie, Ontario                      NaN 

En affichant ma variable ci-haut, j'ai pu vérifier que je l'avais bien faite. 

In [17]:
subs_contrat[pCan].nom_programme.value_counts()

Fonds du Canada pour les périodiques                                    824
Développement de Communautés de Langue Officielle                       461
Développement des communautés par le biais des arts et du patrimoine    447
Célébration, commémoration et apprentissage                             328
Fonds du livre du Canada                                                298
Programme du multiculturalisme                                          213
Fonds du Canada pour la présentation des arts                           176
Programme de soutien au sport                                           125
Mise en valeur des langues officielles                                  120
Programme des peuples autochtones                                       114
Fonds du Canada pour les espaces Culturels                              105
Sport - Programme d'accueil                                              75
Programme d'aide aux musées (PAM)                                        74
Fonds du Can

Ce procédé ci-haut permet de savoir quel programme a octroyé le plus de subventions. Dans ce cas-ci la réponse est : Fonds du Canada pour les périodiques.

In [18]:
prog_subs = subs_contrat[pCan].nom_programme == "Fonds du Canada pour les périodiques"

In [19]:
subs_contrat[pCan][prog_subs]

numero_reference  numero_amendement date_amendement type_entente type_beneficiaire numero_compagnie_beneficiaire                             nom_legal_beneficiaire nom_operationnel_beneficiaire nom_organisation_de_recherche pays_beneficiaire province_beneficiaire                   ville_beneficiaire code_postal_beneficiaire nom_du_comte_federal  numero_du_comte_federal                         nom_programme                                   but_du_programme                                      titre_entente numero_entente  valeur_entente type_monnaie_etrangeres valeur_monaie_etrangere date_debut_entente date_fin_entente couverture                           description  identifiant_SCIAN                                   resultat_attendu                        information_additionnelle abbreviation_ministère                    ministere_responsable
211894  016-2018-2019-Q1-1318116                  0             NaN            C               NaN                           NaN                                 HEBDOS QUÉBEC INC.                           NaN                           NaN                CA                    QC                        Laval | Laval                  H7T 2L1                  NaN                      nan  Fonds du Canada pour les périodiques  Fonds du Canada pour les périodiques: Initiati...  Création d'une plateforme numérique centrale d...        1318116       302658.00                     CAD             $302,658.00         2018-05-15       2018-10-15        NaN  Fonds du Canada pour les périodiques                nan  Assurer un financement aux associations pour d...                                              NaN                    pch  Canadian Heritage | Patrimoine canadien
211895  016-2018-2019-Q1-1318210                  0             NaN            C               NaN                           NaN                                 ALTERNATIVES, INC.                           NaN                           NaN                CA                    ON                Kitchener | Kitchener                  N2H 3X7                  NaN                      nan  Fonds du Canada pour les périodiques  Fonds du Canada pour les périodiques: Innovati...  Empowering the evolution of our digital story-...        1318210        25500.00                     CAD              $25,500.00         2018-05-01       2018-10-01        NaN  Fonds du Canada pour les périodiques                nan  Fournir un financement pour l’innovation comme...                                              NaN                    pch  Canadian Heritage | Patrimoine canadien
211896  016-2018-2019-Q1-1318211                  0             NaN            C               NaN                           NaN              2519098 ONTARIO INC- PHOTOED MAGAZINE                           NaN                           NaN                CA                    ON                    Toronto | Toronto                  M4Y 2W1                  NaN                      nan  Fonds du Canada pour les périodiques  Fonds du Canada pour les périodiques: Innovati...  PhotoEd Magazine - Outreach events + FSI campaign        1318211        27750.00                     CAD              $27,750.00         2018-06-01       2019-03-01        NaN  Fonds du Canada pour les périodiques                nan  Fournir un financement pour l’innovation comme...                                              NaN                    pch  Canadian Heritage | Patrimoine canadien
211897  016-2018-2019-Q1-1318335                  0             NaN            C               NaN                           NaN                       ÉDITIONS PROTÉGEZ-VOUS (LES)                           NaN                           NaN                CA                    QC                  Montréal | Montréal                  H2K 1C3                  NaN                      nan  Fonds du Canada pour les périodiques  Fonds du Canada pour les périodiques: Innovati...  Mise en place d'un programme numérique

Ci-haut se trouve tous les contrats/subventions octroyés par le programme Fonds du Canada pour les périodiques.

In [20]:
subs_contrat[pCan][prog_subs].nom_legal_beneficiaire.value_counts()

SALTWIRE NETWORK INC. -  NEWFOUNDLAND                                    11
SALTWIRE NETWORK INC.- NOVA SCOTIA                                        7
GLACIER FARMMEDIA LIMITED PARTNERSHIP                                     5
IQ BUSINESS MEDIA INC.                                                    4
RAVEN MÉDIA INC.                                                          4
MAGAZINES CANADA                                                          4
PEGASUS PUBLICATIONS INC.                                                 3
CANADIAN NEWS MEDIA ASSOCIATION                                           3
CANADIAN SCRAPBOOKER INC.                                                 3
ALTERNATIVES, INC.                                                        3
MÉDIAS TRANSCONTINENTAL S.E.N.C. / TRANSCONTINENTAL MEDIA G.P.            3
GRASSLANDS NEWS GROUP                                                     3
ECOPARENT INC.                                                            3
OUTDOOR GROU

Bon, je sais qu'il ne fallait pas afficher les récurrences des différents bénéficiaires. Toutefois en faisant cette erreur au départ, j'ai vu cette maison d'édition à Winnipeg, Manitoba : PEGASUS PUBLICATIONS INC. et étant un fan de Marvel voici à quoi j'ai pensé. https://marvelcinematicuniverse.fandom.com/wiki/Project_P.E.G.A.S.U.S. (J'espère ne pas _spoil_ au sujet de Captain Marvel :/)

In [21]:
subs_contrat[pCan][prog_subs].groupby("nom_legal_beneficiaire").valeur_entente.sum().sort_values(ascending=False)

nom_legal_beneficiaire
TVA PUBLICATIONS INC. - CANADIAN LIVING                  1500000.00
TVA PUBLICATIONS INC. - TV HEBDO                         1500000.00
ROGERS MEDIA INC. - MACLEAN'S                            1403958.00
TVA PUBLICATIONS INC. - STYLE AT HOME                    1368536.00
TVA PUBLICATIONS INC. - 7 JOURS                          1306990.00
FEATURE PUBLISHING LTD.                                  1264919.00
HOUSE & HOME MEDIA - CANADIAN HOUSE & HOME               1264584.00
READER'S DIGEST MAGAZINES LIMITED - READER'S DIGEST      1247578.00
WESTERN PRODUCER PUBLICATIONS LIMITED PARTNERSHIP        1207816.00
TVA PUBLICATIONS INC. - COUP DE POUCE                    1199271.00
PUBLICATIONS CHARRON & CIE INC. (LES) - LA SEMAINE       1113622.00
FÉDÉRATION DE L'ÂGE D'OR DU QUÉBEC                       1070534.00
GLACIER FARMMEDIA LIMITED PARTNERSHIP                    1049976.00
ZOOMERMEDIA LIMITED                                      1045168.00
ROGERS MEDIA INC. - CHATE

Deux sections de TVA Publications INC. (Canadian Living et Tv Hebdo) ont chacun reçu 1 500 000.00 $

In [22]:
subs_contrat[pCan][prog_subs]

numero_reference  numero_amendement date_amendement type_entente type_beneficiaire numero_compagnie_beneficiaire                             nom_legal_beneficiaire nom_operationnel_beneficiaire nom_organisation_de_recherche pays_beneficiaire province_beneficiaire                   ville_beneficiaire code_postal_beneficiaire nom_du_comte_federal  numero_du_comte_federal                         nom_programme                                   but_du_programme                                      titre_entente numero_entente  valeur_entente type_monnaie_etrangeres valeur_monaie_etrangere date_debut_entente date_fin_entente couverture                           description  identifiant_SCIAN                                   resultat_attendu                        information_additionnelle abbreviation_ministère                    ministere_responsable
211894  016-2018-2019-Q1-1318116                  0             NaN            C               NaN                           NaN                                 HEBDOS QUÉBEC INC.                           NaN                           NaN                CA                    QC                        Laval | Laval                  H7T 2L1                  NaN                      nan  Fonds du Canada pour les périodiques  Fonds du Canada pour les périodiques: Initiati...  Création d'une plateforme numérique centrale d...        1318116       302658.00                     CAD             $302,658.00         2018-05-15       2018-10-15        NaN  Fonds du Canada pour les périodiques                nan  Assurer un financement aux associations pour d...                                              NaN                    pch  Canadian Heritage | Patrimoine canadien
211895  016-2018-2019-Q1-1318210                  0             NaN            C               NaN                           NaN                                 ALTERNATIVES, INC.                           NaN                           NaN                CA                    ON                Kitchener | Kitchener                  N2H 3X7                  NaN                      nan  Fonds du Canada pour les périodiques  Fonds du Canada pour les périodiques: Innovati...  Empowering the evolution of our digital story-...        1318210        25500.00                     CAD              $25,500.00         2018-05-01       2018-10-01        NaN  Fonds du Canada pour les périodiques                nan  Fournir un financement pour l’innovation comme...                                              NaN                    pch  Canadian Heritage | Patrimoine canadien
211896  016-2018-2019-Q1-1318211                  0             NaN            C               NaN                           NaN              2519098 ONTARIO INC- PHOTOED MAGAZINE                           NaN                           NaN                CA                    ON                    Toronto | Toronto                  M4Y 2W1                  NaN                      nan  Fonds du Canada pour les périodiques  Fonds du Canada pour les périodiques: Innovati...  PhotoEd Magazine - Outreach events + FSI campaign        1318211        27750.00                     CAD              $27,750.00         2018-06-01       2019-03-01        NaN  Fonds du Canada pour les périodiques                nan  Fournir un financement pour l’innovation comme...                                              NaN                    pch  Canadian Heritage | Patrimoine canadien
211897  016-2018-2019-Q1-1318335                  0             NaN            C               NaN                           NaN                       ÉDITIONS PROTÉGEZ-VOUS (LES)                           NaN                           NaN                CA                    QC                  Montréal | Montréal                  H2K 1C3                  NaN                      nan  Fonds du Canada pour les périodiques  Fonds du Canada pour les périodiques: Innovati...  Mise en place d'un programme numérique

In [23]:
subs_contrat[pCan][prog_subs].province_beneficiaire.value_counts()

ON    323
QC    191
BC     81
AB     70
SK     51
MB     43
NS     24
NL     18
NB     12
NT      6
PE      4
YT      1
Name: province_beneficiaire, dtype: int64

In [24]:
subs_contrat[pCan][prog_subs].province_beneficiaire.value_counts() / len(subs_contrat[pCan][prog_subs])*100

ON   39.20
QC   23.18
BC    9.83
AB    8.50
SK    6.19
MB    5.22
NS    2.91
NL    2.18
NB    1.46
NT    0.73
PE    0.49
YT    0.12
Name: province_beneficiaire, dtype: float64

Voici donc le pourcentage des subventions reçues pour chacune des provinces au niveau du Fonds du Canada pour les périodiques. Sur un total de 824, QC avec 191 compte pour 23,18%. 